site/content/how-to/monitoring/troubleshooting.md

```diff
@@ -65,54 +68,321 @@ LAST SEEN   TYPE      REASON              OBJECT
 5s          Warning   ResourceDeleted     nginxgateway/ngf-config                          NginxGateway configuration was deleted; using defaults
 ```
 
-##### Logs
+#### Get shell access to NGINX container
+
+Getting shell access to containers allows developers and operators to view the environment of a running container, see its logs or diagnose any problems. To get shell access to the NGINX container, use `kubectl exec`:
+
+```shell
+kubectl exec -it -n nginx-gateway  <ngf-pod-name> -c nginx /bin/sh
+```
+
+#### Logs
+
+Logs from the NGINX Gateway Fabric control plane and data plane can contain information that isn't available to status or events. These can include errors in processing or passing traffic. You can see logs for a crashed or killed container by adding the `-p` flag to the above commands.
+
+1. Container Logs
+
+    To see logs for control plane container:
+
+    ```shell
+    kubectl -n nginx-gateway logs <ngf-pod-name> -c nginx-gateway
+    ```
+
+    To see logs for data plane container:
+
+    ```shell
+    kubectl -n nginx-gateway logs <ngf-pod-name> -c nginx
+    ```
+
+1. Error Logs
+
+    For _nginx-gateway_ container, you can `grep` for the word `error` or change the log level to `error` by following steps in Modify log levels (Section 4).
```

Comment:
I still think this is confusing. What if we just had:

```suggestion
    For _nginx-gateway_ container, you can `grep` the logs for the word `error`:
```

Then in the section on modify log levels, describe that you can change the log level to `error` to see only the error log messages from the control plane. The modify log level section can also include the instructions on greping for debug log messages.

Reply:
We do already have a section on error logs maybe we can say how to modify log levels and have a `note` on how to grep debug logs?

Reply:
I'm just suggesting that you keep all the info related or dependent on modifying the log level in the modify log level section. IMO it's confusing when the steps depend/refer to one another.

Reply:
Okay, the dependency makes sense. I will put them all together.